internal/util-logging/src/main/scala/sbt/internal/util/ConsoleAppender.scala

```diff
@@ -370,18 +372,25 @@ class ConsoleAppender private[ConsoleAppender] (
       s"$DeleteLine  | => ${item.name} ${elapsed}s"
     }
 
+    val width = ConsoleAppender.terminalWidth
+    val extra: Int = info.foldLeft(0)(_ + terminalLines(width)(_))
     val previousLines = progressLines.getAndSet(info)
+    val prevExtra = previousLines.foldLeft(0)(_ + terminalLines(width)(_))
+
     val prevPadding = padding.get
-    val newPadding = math.max(0, previousLines.length + prevPadding - info.length)
+    val newPadding =
+      math.max(0, previousLines.length + prevExtra + prevPadding - info.length - extra)
     padding.set(newPadding)
 
     deleteConsoleLines(newPadding)
     deleteConsoleLines(blankZone)
     info.foreach(i => out.println(i))
 
-    out.print(cursorUp(blankZone + info.length + newPadding))
+    out.print(cursorUp(blankZone + info.length + newPadding + extra))
     out.flush()
   }
+  private def terminalLines(width: Int): String => Int =
+    (progressLine: String) => (progressLine.length - 1) / width
```

Comment:
I wonder if `width` here could ever be zero for a terminal that someone enables super shell (due to JLine or otherwise).

Reply:
Good question. That didn't occur to me. Regardless of whether or not we can repro, I think a zero check makes sense.